miden/tests/integration/operations/decorators/advice.rs

```diff
@@ -117,3 +117,20 @@ fn advice_inject_u64div_conditional_execution() {
     let test = build_test!(source, &[8, 0, 4, 0, 1, 0]);
     test.expect_stack(&[0, 0, 0, 0, 0, 4, 0, 8]);
 }
+
+#[test]
+fn advice_inject_mem() {
+    let source = "begin
+    # write to memory and drop values from stack
+    mem_storew.2 dropw mem_storew.3 dropw
+    # copy from memory to advice map
+    adv.mem.2.2
+    # copy from advice map to advice tape
+    adv.keyval
+    # copy from advice tape to stack
+    adv_loadw swapw adv_loadw swapw
```

Comment:
I found the last `swapw` counterintuitive and spent some time trying to figure out why it is there. It would be good to explain in comments what is the expected state after some of the steps. For example, originally, I assumed that `[1, 2, 3, 4]` is saved into `mem[2]` and `[5, 6, 7, 8]` is stored into `mem[3]` - but it seems to be the opposite?

Reply:
I think it's the same just with the elements in the reverse order, ie, `mem[2]` should be `[4, 3, 2, 1]` and `mem[3]` should be `[8, 7, 6, 5]`. The last swap is to bring the order back to the original one as before the swap the stack is `[5, 6, 7, 8, 1, 2, 3, 4]`. I've added comments in the test now. Please let me know if anything is unclear now or if I made a mistake somewhere.

Reply:
The comments look great! Thank you! Though, I think we can simplify them to list only the component states which have changed.

Reply:
Done.